meson.build

```diff
@@ -39,9 +42,10 @@ if target_machine.system() == 'freebsd'
 endif
 
 xapian_dep = dependency('xapian-core',
-                        required:false,
+                        required:get_option('with_xapian'),
                         static:static_linkage)
-conf.set('ENABLE_XAPIAN', xapian_dep.found())
+private_conf.set('ENABLE_XAPIAN', xapian_dep.found())
+public_conf.set('LIBZIM_WITH_XAPIAN', xapian_dep.found())
```

Comment:
xapian will be enabled even if `-Dwith_xapian=false` is requested but xapian is available

Reply:
Indeed, I was using the `feature` option but it is not available on too old version of meson (ubuntu bionic) so I change the option to boolean and forget this part.

Fixed with last commit (to be squashed before merge).